### PR TITLE
Add PasswordBox widget

### PR DESCRIPTION
**Pull request by WinXaito**
This PR add a new widget: `PasswordBox` : https://learn.microsoft.com/en-us/windows/apps/design/controls/password-box

This looks like that:

passwordbox

There is four modes:

- `peek` : default mode, when the user got the focus (for the first time !) and the text field is not empty, a reveal button is added. If the focus is lost and re-got, the button is not visible (same operation than Microsoft WinUI). The button is visible again only if the text field is cleared.
- `peekAlways`: (not a Microsoft standard mode). The button is visible when the text-field is not empty and doesn't look if it the first focus or not).
- `visible` : the password is always visible (useful for a programmatically change)
- `hidden`: the password is always hidden (useful for a programmatically change)

-----------------

This PR also add parameters `onTapDown` and `onTapUp` on all buttons.

## Pre-launch Checklist

- [x] I have updated `CHANGELOG.md` with my changes  
- [x] I have run "dart format ." on the project  
- [x] I have added/updated relevant documentation

### CODE REVIEW

**Comment by WinXaito** [comment]
There is one think, I didn't found this icon: 
image

So I used this one for the moment:

image

**Review by bdlukaa — COMMENTED** [review]
LGTM overall

File: lib/src/controls/form/password_box.dart [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+      _internalNode ??= _createFocusNode();
+    }
+    controller.addListener(_handleTextChange);
+    focusNode!.addListener(_handleFocusChange);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
```
[the remark this answers is not in the document]
Comment (WinXaito) [reply lib/src/controls/form/password_box.dart:184]: So, does this must be the case for the FocusNode too ? (If this is the case, I will must update the NumberBox too)

[the remark this answers is not in the document]
Comment (bdlukaa) [reply lib/src/controls/form/password_box.dart]: I think so

File: lib/src/controls/form/password_box.dart [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+          textCanPeek);
+
+  @override
+  void initState() {
+    if (widget.focusNode == null) {
+      _internalNode = FocusNode(debugLabel: '${widget.runtimeType}');
+    }
+    if (widget.controller == null) {
+      _internalController = TextEditingController();
+    }
```
Comment (bdlukaa) [inline lib/src/controls/form/password_box.dart:180]: We also must do this check on `didUpdateWidget`. It is a common case when debugging to attach a controller/node after the widget was already built (hot reload)

File: lib/src/controls/form/password_box.dart [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+/// A fluent design input form for password.
+///
+/// A PasswordBox lets the user enter a password. There is multiple mode to
+/// indicate if the password must be hidden, visible or if a reveal button is
+/// shown.
+///
+/// See also:
+///
+///  * https://learn.microsoft.com/en-us/windows/apps/design/controls/password-box
```
Comment (bdlukaa) [inline lib/src/controls/form/password_box.dart:37]:
```suggestion
/// A fluent design input form for password.
///
/// A password box is a text input box that conceals the characters typed into
/// it for the purpose of privacy. A password box looks like a text box, except 
/// that it renders placeholder characters in place of the text that has been 
/// entered. You can configure the placeholder character.
///
/// ![PasswordBox](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/passwordbox-focus-typing.png)
///
/// See also:
///
///  * <https://learn.microsoft.com/en-us/windows/apps/design/controls/password-box>
///  * [PasswordRevealMode], the different modes that the password box can have
///  * [TextBox], the underlaying widget that renders the text box
```

File: lib/src/controls/form/password_box.dart [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+class PasswordBox extends StatefulWidget {
+  /// Controls the text being edited.
+  ///
+  /// If null, this widget will create its own [TextEditingController].
+  final TextEditingController? controller;
+
+  /// Disables the text field when false.
+  ///
+  /// Text fields in disabled states have a light grey background and don't
+  /// respond to touch events including the [prefix], [suffix].
```
Comment (bdlukaa) [inline lib/src/controls/form/password_box.dart:47]: We don't have prefix nor suffix

```suggestion
  /// respond to touch events.
```

**Review by bdlukaa — APPROVED** [review]

**Pull request merged**